src/ocean/meta/traits/Basic.d

```diff
@@ -0,0 +1,424 @@
+        || is( Unqual!(T) == real );
+}
+
+///
+unittest
+{
+    static assert ( isRealType!(double));
+    static assert (!isRealType!(long));
+    static assert (!isRealType!(cdouble));
+}
+
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is a complex floating point type
+
+*******************************************************************************/
+
+public template isComplexType( T )
+{
+    const bool isComplexType =
+           is( Unqual!(T) == cfloat )
+        || is( Unqual!(T) == cdouble )
+        || is( Unqual!(T) == creal );
+}
+
+///
+unittest
+{
+    static assert ( isComplexType!(cdouble));
+    static assert ( isComplexType!(Const!(cdouble)));
+    static assert (!isComplexType!(double));
+}
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is an imaginary floating point type
+
+*******************************************************************************/
+
+public template isImaginaryType( T )
+{
+    const bool isImaginaryType =
+           is( Unqual!(T) == ifloat )
+        || is( Unqual!(T) == idouble )
+        || is( Unqual!(T) == ireal );
+}
+
+///
+unittest
+{
+    static assert ( isImaginaryType!(idouble));
+    static assert ( isImaginaryType!(Const!(idouble)));
+    static assert (!isImaginaryType!(double));
+}
+
+/*******************************************************************************
+
+    Params:
+        T = type to check
+
+    Returns:
+        `true` if `T` is any floating-point type: real, complex, or
+        imaginary.
+
+*******************************************************************************/
+
+public template isFloatingPointType( T )
+{
+    const bool isFloatingPointType =
+           isRealType!(T)
+        || isComplexType!(T)
+        || isImaginaryType!(T);
+}
+
+///
+unittest
+{
+    static assert (isFloatingPointType!(double));
+    static assert (isFloatingPointType!(ifloat));
+}
+
+/*******************************************************************************
+
+    Used by other traits to distinguish between dynamic and static arrays
+    instead of plain `bool` values.
+
+*******************************************************************************/
+
+public enum ArrayKind
+{
+    NotArray    = 0,
+    Static      = 1,
+    Dynamic     = 2,
+    Associative = 3,
+}
+
+/*******************************************************************************
+
+    Check if type is an array type and which kind of array it is
+
+    Params:
+        T = type to check
+
+    Returns:
+        `ArrayKind` value indicating if `T` is an array and if it is, static or
+        dynamic
+
+*******************************************************************************/
+
+public template isArrayType ( T )
+{
+    static if (is(T U == U[]))
```

Comment:
My point is that relying on a compiler bug may potentially be a suboptimal design decision. Do you prefer code that relies on compiler bugs for convenience?
Maybe this bug will never be fixed, maybe it will. Maybe it will declared to be a feature as you predict, but that would at least cause more `is` expression inconsistencies with type qualifiers. A perfectly consistent solution would be to just ignore all qualifiers in all `is(A == B)` and `is(A == B C)` expressions, which would make me very happy.